reconcile/gitlab_housekeeping.py

```diff
@@ -179,6 +180,71 @@ def clean_pipelines(
                 )
 
 
+def verify_ondemend_tests(
+    dry_run: bool,
+    mr: ProjectMergeRequest,
+    must_pass: Iterable[str],
+    gl: GitLabApi,
+    gl_instance: Mapping[str, Any],
+    gl_settings: Mapping[str, Any],
+    state: State,
+) -> bool:
+    pipelines = gl.get_merge_request_pipelines(mr)
+    running_pipelines = [p for p in pipelines if p["status"] == "running"]
+    if running_pipelines:
+        # wait for pipeline complate
+        return False
+
+    gl_fork = GitLabApi(
+        instance=gl_instance,
+        project_id=mr.source_project_id,
+        settings=gl_settings,
+    )
```

Comment:
any reason to create a new `GitLabApi` instead of reuse current `gl`, `GitLabApi` instance require `cleanup`.

Reply:
The original gl targets the upstream repo while this gl_fork targets the fork repo. We can only get commit status form the fork repo.

Reply:
you need this part https://github.com/app-sre/qontract-reconcile/blob/dde29ad691e1fd073fe437af071ec3657e5b7097/reconcile/utils/gitlab_api.py#L128-L129, right?

maybe worth extract another method `get_project_by_id`, and replace prevous project_url part with current `get_project`. This can save calls used to init `GitLabApi`, and save those paramter passing of `gl_instance` and `gl_settings`.

Also need to make request counter correct as `commits.get(commit.id)` has 1 request, and  `.statuses.list()` has another (if no pagination), need a better way to count mentioned in https://issues.redhat.com/browse/APPSRE-10450

Reply:
created `get_project_by_id` for both `verify_on_demand_tests` and `clean_pipelines`.
For the request counter, I don't have a good idea of counting it outside of GitLabApi. Can I leave it to you?

Reply:
sure, I can handle that later